src/GPLVM/PCA.hs

```diff
@@ -0,0 +1,44 @@
+module GPLVM.PCA where
+
+import Universum hiding (All, Vector, transpose)
+
+import GPLVM.Types
+
+import Control.Lens (makeLenses)
+import Data.Array.Repa
+import Data.Array.Repa.Repr.ForeignPtr
+import Numeric.LinearAlgebra.Repa hiding (Matrix, Vector)
+
+data PCA = PCA
+  {   _inputData    :: Matrix D Double
+    , _covariance   :: Covariance Double
+    , _eigenVectors :: EigenVectors Double
+    , _eigenValues  :: Vector F Double
+    , _finalData    :: Matrix D Double
+    , _restoredData :: Matrix D Double
```

Comment:
can we remove this field with restored data?

Reply:
Why? It is one of the main results of the algorithm.